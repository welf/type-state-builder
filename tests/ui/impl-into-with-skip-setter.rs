// Test that impl_into and skip_setter attributes conflict

use type_state_builder::TypeStateBuilder;

#[derive(TypeStateBuilder)]
struct ConflictingFieldAttributes {
    #[builder(impl_into, skip_setter, default = "String::new()")]
    name: String,

    age: Option<u32>,
}

fn main() {}
