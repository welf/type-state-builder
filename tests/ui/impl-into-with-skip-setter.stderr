error: Field-level impl_into is incompatible with skip_setter
       note: #[builder(impl_into)] and #[builder(skip_setter)] are incompatible
       help: remove one of these attributes
 --> tests/ui/impl-into-with-skip-setter.rs:5:10
  |
5 | #[derive(TypeStateBuilder)]
  |          ^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the derive macro `TypeStateBuilder` (in Nightly builds, run with -Z macro-backtrace for more info)
